### PR TITLE
First pass ts conversion for lib/mapfiles

### PR DESCRIPTION
**Pull request by jeremy-rifkin**

### CODE REVIEW

File: lib/mapfiles/map-file.ts [hunk 1]
```diff
@@ -26,30 +26,73 @@ import fs from 'fs';
...
+    displayName?: string;
+};
+
+type ReconstructedSegment = {
+    addressInt: number;
+    address: string;
+    endAddress?: string;
+    segmentLength: number;
+    unitName?: string | false;
+};
```
Comment (jeremy-rifkin) [inline lib/mapfiles/map-file.ts:53]: I'm not in love with the types yet, but they work

File: lib/mapfiles/map-file.ts [hunk 2]
```diff
@@ -26,30 +26,73 @@ import fs from 'fs';
...
+
 export class MapFileReader {
+    preferredLoadAddress = 0x400000;
+    segmentMultiplier = 0x1000;
+    segmentOffsets: SegmentOffset[] = [];
+    segments: Segment[] = [];
+    isegments: Segment[] = [];
+    namedAddresses: Segment[] = [];
+    entryPoint: '' | EntryPoint = '';
```
Comment (jeremy-rifkin) [inline lib/mapfiles/map-file.ts:82]: `entryPoint` appears unused but I'm trying to keep changes minimal

File: lib/mapfiles/map-file.ts [hunk 3]
```diff
@@ -60,13 +103,7 @@ export class MapFileReader {
...
 
-    /**
-     *
-     * @param {(string|boolean)} segment
-     * @param {number} address
-     * @returns {(object | boolean)}
-     */
-    getLineInfoByAddress(segment, address) {
+    getLineInfoByAddress(segment: string | boolean, address: number) {
         for (let idx = 0; idx < this.lineNumbers.length; idx++) {
```
Comment (jeremy-rifkin) [inline lib/mapfiles/map-file.ts:107]: Lots of index-based loops in this code but again trying to keep changes minimal

Comment (partouf) [reply lib/mapfiles/map-file.ts:107]: It was written a long time ago in 2017, I just did what I knew would work :D

Comment (jeremy-rifkin) [reply lib/mapfiles/map-file.ts:107]: Perfect for 2017! :P

File: lib/mapfiles/map-file.ts (hunk 1 above)
Comment (jeremy-rifkin) [reply lib/mapfiles/map-file.ts:53]: @partouf  do these types look ok to you?

**Comment by dkm** [comment]
this is live :rocket:

**Pull request merged**